src/cpp/rtps/xmlparser/XMLDynamicParser.cpp

```diff
@@ -343,9 +343,20 @@ XMLP_ret XMLParser::parseXMLAliasDynamicType(
             const char* name = p_root->Attribute(NAME);
             if (name != nullptr)
             {
-                p_dynamictypebuilder_t typeBuilder =
-                        types::DynamicTypeBuilderFactory::get_instance()->create_alias_builder(valueBuilder, name);
-                XMLProfileManager::insertDynamicTypeByName(name, typeBuilder);
+                if (nullptr == XMLProfileManager::getDynamicTypeByName(name))
+                {
+                    p_dynamictypebuilder_t typeBuilder =
+                            types::DynamicTypeBuilderFactory::get_instance()->create_alias_builder(valueBuilder, name);
+                    if (false == XMLProfileManager::insertDynamicTypeByName(name, typeBuilder))
+                    {
+                        types::DynamicTypeBuilderFactory::get_instance()->delete_builder(typeBuilder);
+                    }
```

Comment:
Shell we return an error in case `XMLProfileManager::insertDynamicTypeByName` return false? 
```suggestion
                    }
                    else
                    {
                     ret = XMLP_ret::XML_ERROR;
                    }
```